### PR TITLE
Seperate market product group and trader risk group contexts

### PR DESCRIPTION
**Pull request by Clish254**
## Description
This separates the market product group and trader risk group contexts for modularity and also in an attempt to improve performance.
## Types of changes

- [ ] Technical Debt (non-breaking change which removes unused code/assets)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:

- [ ] The related ClickUp task has been linked to this PR
- [ ] The person creating the pull request is listed in "Assignees"
- [ ] Change requires updated documentation

### CODE REVIEW

File: src/context/market_product_group.tsx [hunk 1]
```diff
@@ -0,0 +1,170 @@
...
+  const [activeProduct, setActiveProduct] = useState<IActiveProduct>(MAINNET_MPs[0])
+  const { isDevnet } = useCrypto()
+  const [initTesting, setInitTesting] = useState<boolean>(false)
+  const [perpInstanceSdk, setPerpInstanceSdk] = useState<Perp>(null)
+  const [perpProductInstanceSdk, setPerpProductInstanceSdk] = useState<Product>(null)
+
+  const wallet: WalletContextState = useWallet()
+
+  const { perpsConnection: mainnetConnection } = useConnectionConfig()
+  const MPG_ID = useMemo(() => (isDevnet ? DEVNET_MPG_ID : MAINNET_MPG_ID), [isDevnet])
```
Comment (AlexOfc) [inline src/context/market_product_group.tsx]: Memo not needed here, it is overhead for simple calculations

File: src/context/market_product_group.tsx [hunk 2]
```diff
@@ -0,0 +1,170 @@
...
+
+  const MPs = useMemo(() => (isDevnet ? DEVNET_MPs : MAINNET_MPs), [isDevnet])
```
Comment (AlexOfc) [inline src/context/market_product_group.tsx]: same

File: src/context/market_product_group.tsx [hunk 3]
```diff
@@ -0,0 +1,170 @@
...
+
+  const { on, off } = useSolSub()
+  const connection = useMemo(() => mainnetConnection, [])
```
Comment (AlexOfc) [inline src/context/market_product_group.tsx]: It is already memoized

File: src/context/market_product_group.tsx [hunk 4]
```diff
@@ -0,0 +1,170 @@
...
+
+  useActivityTracker({
+    callbackOff: () => setIsCurrentTabActive(false),
+    callbackOn: () => setIsCurrentTabActive(true)
+  })
+  const [isCurrentTabActive, setIsCurrentTabActive] = useState(true)
```
Comment (AlexOfc) [inline src/context/market_product_group.tsx]: state has to be on the top of component

File: src/context/market_product_group.tsx [hunk 5]
```diff
@@ -0,0 +1,170 @@
...
+    //if (!price || Number.isNaN(+price) || Number.isNaN(+displayFractional(res))) return null
+    //const percent = (+displayFractional(res) / +price) * 100
+    //setFundingRate(percent.toFixed(4))
+  }
+
+  useEffect(() => {
+    ;(async () => {
+      if (!currentMPG) return
+      const mpgData = await MarketProductGroup.fetch(connection, currentMPG)
+      updateMPGDetails(mpgData ? mpgData[0] : null, mpgData ? mpgData[1] : null)
```
Comment (AlexOfc) [inline src/context/market_product_group.tsx]: destructruct is ok here

File: src/context/market_product_group.tsx [hunk 6]
```diff
@@ -0,0 +1,170 @@
...
+  const { on, off } = useSolSub()
+  const connection = useMemo(() => mainnetConnection, [])
+
+  useActivityTracker({
+    callbackOff: () => setIsCurrentTabActive(false),
+    callbackOn: () => setIsCurrentTabActive(true)
+  })
+  const [isCurrentTabActive, setIsCurrentTabActive] = useState(true)
+
+  const updateMPGDetails = async (mpgData, mpgRawData) => {
```
Comment (AlexOfc) [inline src/context/market_product_group.tsx:68]: U can put this function in useEffect to prevent reallocation

**Review by AlexOfc — CHANGES_REQUESTED** [review]

File: src/context/market_product_group.tsx (hunk 6 above)
Comment (Clish254) [reply src/context/market_product_group.tsx:68]: This is used in two useEffects

**Review by deadeagle63 — APPROVED** [review]

**Review by walt-1 — APPROVED** [review]

**Comment by walt-1** [comment]
@Clish254 - when i disconnect my wallet, the account page info does not reset nor does the positions panel on the trade page.

Can you patch that on this PR?

**Comment by Clish254** [comment]
> @Clish254 - when i disconnect my wallet, the account page info does not reset nor does the positions panel on the trade page.
> 
> Can you patch that on this PR?

Yes sure

**Comment by Clish254** [comment]
> @Clish254 - when i disconnect my wallet, the account page info does not reset nor does the positions panel on the trade page.
> 
> Can you patch that on this PR?

I have added a commit for this @walt-1

**Review by walt-1 — APPROVED** [review]

**Pull request merged**